api/v1/installation_types.go

```diff
@@ -652,6 +652,9 @@ type InstallationStatus struct {
 	// +optional
 	Computed *InstallationSpec `json:"computed,omitempty"`
 
+	// CalicoVersion shows the current running version of calico.
+	CalicoVersion string `json:"calicoVersion,omitempty"`
```

Comment:
This field was added so at the end of the core_controllers' reconcile loop the version is set. This way, you know if calico-system has rolled out successfully and that you can go ahead with deploying other things, in this case EGW. This is better than simply using the Ready state, since the EGW controller might win the "race" when an operator pod is just starting after an upgrade and that Ready state might be tied to the old version being rolled out.

For this reason it is also important that the contents of this field reflect the OSS version in case of Variant OSS and Enterprise in case of Calico Enterprise. With the current implementation if we take branch 1.27 as an example the version will be either:
- v3.14.3 for EE
- v3.23.5 for OSS
@tmjd I was wondering if this is confusing since there are v3.x.x version overlaps between OSS and EE. I was wondering if we should prefix or postfix it with an indicator or if this is ok, since the variant is also part of the status. What is your opinion on this?

I would say that if we keep the implementation the way it is right now, we should at least improve the comment of this field.

Reply:
I don't think it is confusing enough to duplicate the Variant info by including a prefix or postfix on the CalicoVersion. I do think it would be good to make it clear in the comments that both the Variant and CalicoVersion are needed to know the exact version deployed.